src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -0,0 +1,43 @@
+module Staging
+  class StagingProjectCreator
+    def initialize(request_body, staging_workflow)
+      @request_body = request_body
+      @staging_workflow = staging_workflow
+    end
+
+    def call
+      if @request_body.blank?
+        errors << 'Staging projects are empty'
+        return self
+      end
+      staging_projects = Xmlhash.parse(@request_body).elements('staging_project').collect do |name|
```

Comment:
make sure the controller adds a validation on the input - otherwise the parse here might return nil.

The good news: we already have a validator builtin. Use validate_action

Reply:
done, can you please check it @coolo